pdns/rec_channel.cc

```diff
@@ -17,7 +17,7 @@
 
 #include "namespaces.hh"
 
-volatile sig_atomic_t RecursorControlChannel::stop = 0;
+std::atomic<bool> RecursorControlChannel::stop = false;
```

Comment:
Perhaps also update the `RecursorControlChannel::stop = 1;` line in `rec_channel_rec.cc`, in the hope of one day enabling `-Wint-in-bool-context` or equivalent?